File: demos/java/dsa/linear/DynamicArrayDemo.java
public class DynamicArrayDemo {
    public static void main(String[] args) {
        System.out.println("=== Java ArrayList & Vector Demo ===");
        java.util.List<Integer> list = new java.util.ArrayList<>();
        list.add(10);
        list.add(20);
        list.add(30);

        if (list.size() != 3 || list.get(1) != 20) {
            throw new RuntimeException("Assertion failed");
        }

        int sum = list.stream().mapToInt(Integer::intValue).sum();
        if (sum != 60) {
            throw new RuntimeException("Sum assertion failed");
        }

        System.out.println("Java ArrayList size=" + list.size() + ", sum=" + sum);
        System.out.println("Java Dynamic Array tests passed successfully.");
    }
}

File: demos/csharp/dsa/linear/DynamicArrayDemo.cs
using System;
using System.Collections.Generic;

public class DynamicArrayDemo
{
    public static void Main()
    {
        Console.WriteLine("=== C# List<T> & Span<T> Demo ===");
        var list = new List<int> { 10, 20 };
        list.Add(30);

        if (list.Count != 3 || list[1] != 20)
            throw new Exception("List assertion failed");

        list.RemoveAt(list.Count - 1);
        if (list.Count != 2)
            throw new Exception("Remove assertion failed");

        ReadOnlySpan<int> span = list.ToArray().AsSpan();
        int sum = 0;
        foreach (var val in span) sum += val;
        if (sum != 30) throw new Exception("Span sum failed");

        Console.WriteLine($"C# List size={list.Count}, sum={sum}");
        Console.WriteLine("C# Dynamic Array tests passed successfully.");
    }
}
